File: src/main/java/org/apache/fineract/client/services/CreditBureauConfigurationApi.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
package org.apache.fineract.client.services;

import org.apache.fineract.client.CollectionFormats.*;

import rx.Observable;
import retrofit2.http.*;

import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import okhttp3.MultipartBody;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public interface CreditBureauConfigurationApi {
  /**
   * 
   * 
   * @param organisationCreditBureauId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("CreditBureauConfiguration/organisationCreditBureau/{organisationCreditBureauId}")
  Observable<String> addOrganisationCreditBureau(
    @retrofit2.http.Path("organisationCreditBureauId") Long organisationCreditBureauId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param creditBureauId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("CreditBureauConfiguration/configuration/{creditBureauId}")
  Observable<String> createCreditBureauConfiguration(
    @retrofit2.http.Path("creditBureauId") Long creditBureauId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param organisationCreditBureauId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @POST("CreditBureauConfiguration/mappings/{organisationCreditBureauId}")
  Observable<String> createCreditBureauLoanProductMapping(
    @retrofit2.http.Path("organisationCreditBureauId") Long organisationCreditBureauId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @return Observable&lt;String&gt;
   */
  @GET("CreditBureauConfiguration/loanProduct")
  Observable<String> fetchLoanProducts();
    

  /**
   * 
   * 
   * @param loanProductId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("CreditBureauConfiguration/loanProduct/{loanProductId}")
  Observable<String> fetchMappingByLoanProductId(
    @retrofit2.http.Path("loanProductId") Long loanProductId
  );

  /**
   * 
   * 
   * @param organisationCreditBureauId  (required)
   * @return Observable&lt;String&gt;
   */
  @GET("CreditBureauConfiguration/config/{organisationCreditBureauId}")
  Observable<String> getConfiguration(
    @retrofit2.http.Path("organisationCreditBureauId") Long organisationCreditBureauId
  );

  /**
   * 
   * 
   * @return Observable&lt;String&gt;
   */
  @GET("CreditBureauConfiguration")
  Observable<String> getCreditBureau();
    

  /**
   * 
   * 
   * @return Observable&lt;String&gt;
   */
  @GET("CreditBureauConfiguration/mappings")
  Observable<String> getCreditBureauLoanProductMapping();
    

  /**
   * 
   * 
   * @return Observable&lt;String&gt;
   */
  @GET("CreditBureauConfiguration/organisationCreditBureau")
  Observable<String> getOrganisationCreditBureau();
    

  /**
   * 
   * 
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("CreditBureauConfiguration/organisationCreditBureau")
  Observable<String> updateCreditBureau(
    @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param configurationId  (required)
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("CreditBureauConfiguration/configuration/{configurationId}")
  Observable<String> updateCreditBureauConfiguration(
    @retrofit2.http.Path("configurationId") Long configurationId, @retrofit2.http.Body String body
  );

  /**
   * 
   * 
   * @param body  (optional)
   * @return Observable&lt;String&gt;
   */
  @Headers({
    "Content-Type:application/json"
  })
  @PUT("CreditBureauConfiguration/mappings")
  Observable<String> updateCreditBureauLoanProductMapping(
    @retrofit2.http.Body String body
  );

}

File: src/main/kotlin/org/openapitools/client/apis/CreditBureauConfigurationApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path


interface CreditBureauConfigurationApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param organisationCreditBureauId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/CreditBureauConfiguration/organisationCreditBureau/{organisationCreditBureauId}")
    suspend fun addOrganisationCreditBureau(
        @Path("organisationCreditBureauId") organisationCreditBureauId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param creditBureauId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/CreditBureauConfiguration/configuration/{creditBureauId}")
    suspend fun createCreditBureauConfiguration(
        @Path("creditBureauId") creditBureauId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param organisationCreditBureauId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/CreditBureauConfiguration/mappings/{organisationCreditBureauId}")
    suspend fun createCreditBureauLoanProductMapping(
        @Path("organisationCreditBureauId") organisationCreditBureauId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/CreditBureauConfiguration/loanProduct")
    suspend fun fetchLoanProducts(): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param loanProductId
     * @return [kotlin.String]
     */
    @GET("v1/CreditBureauConfiguration/loanProduct/{loanProductId}")
    suspend fun fetchMappingByLoanProductId(@Path("loanProductId") loanProductId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param organisationCreditBureauId
     * @return [kotlin.String]
     */
    @GET("v1/CreditBureauConfiguration/config/{organisationCreditBureauId}")
    suspend fun getConfiguration(@Path("organisationCreditBureauId") organisationCreditBureauId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/CreditBureauConfiguration")
    suspend fun getCreditBureau(): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/CreditBureauConfiguration/mappings")
    suspend fun getCreditBureauLoanProductMapping(): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/CreditBureauConfiguration/organisationCreditBureau")
    suspend fun getOrganisationCreditBureau(): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/CreditBureauConfiguration/organisationCreditBureau")
    suspend fun updateCreditBureau(@Body body: String? = null): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param configurationId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/CreditBureauConfiguration/configuration/{configurationId}")
    suspend fun updateCreditBureauConfiguration(
        @Path("configurationId") configurationId: Long,
        @Body body: String? = null
    ): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/CreditBureauConfiguration/mappings")
    suspend fun updateCreditBureauLoanProductMapping(@Body body: String? = null): String

}
